core/connection.js

```diff
@@ -619,3 +621,32 @@ Blockly.Connection.prototype.setShadowDom = function(shadow) {
 Blockly.Connection.prototype.getShadowDom = function() {
   return this.shadowDom_;
 };
+
+/**
+ * @return {string} A human-readable description of this Connection.
+ */
+Blockly.Connection.prototype.describe = function() {
+  var msg;
+  var block = this.sourceBlock_;
+  if (!block) {
+    return 'Orphan Connection';
+  } else if (block.outputConnection === this) {
+    msg = 'Output Connection of ';
+  } else if (block.previousConnection === this) {
+    msg = 'Previous Connection of ';
+  } else if (block.nextConnection === this) {
+    msg = 'Next Connection of ';
+  } else {
+    for (var i = 0; i < block.inputList.length; ++i) {
+      var input = block.inputList[i];
```

Comment:
That assumption is suspect (though not in this case).  If that was a standard way of looping arrays, I would have expected to find it here: https://jsperf.com/fastest-array-loops-in-javascript/32.

Refactored into goog.array.find(..), which parallels the fastest Array.prototype.find(..) in the above link (which isn't available in IE).